eth/downloader/downloader.go

```diff
@@ -503,22 +504,38 @@ func (d *Downloader) syncWithPeer(p *peerConnection, hash common.Hash, td *big.I
 		// nil panics on an access.
 		pivot = d.blockchain.CurrentBlock().Header()
 	}
-	height := latest.Number.Uint64()
 
-	origin, err := d.findAncestor(p, latest)
+	// If the remote peer is lagging behind, no need to sync with it, drop the peer.
+	remoteHeight := remoteHeader.Number.Uint64()
+	var localHeight uint64
+	switch mode {
```

Comment:
![image](https://user-images.githubusercontent.com/7310198/216876326-788cc4e8-e0e2-488c-8a3a-0e8e69db32c3.png)

I am curious why the condition here does not stop the syncing from a lag peer.

Reply:
there could be 2 scenarios:
1.the Peer broadcast its state with a invalid TD, could be a very big TD, but it does not have the block.
2.local node imported new blocks, between `nextSyncOp` and `syncWithPeer, fetchHead`, it is likely to happen, since block fetch and download are 2 asynchronous operation.